document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -43,11 +49,48 @@ public DocumentInfo getDocumentInfo() {
                 // company accounts should work for abridged, resulting in this abridged specific code
                 // qualifying for removal
                 .filter(this::isAbridged)
-                .map(accountsLinks -> new DocumentInfo())
+                .map(accountsLinks -> getAbridgedAccountData(resourceId))
                 .orElse(null);
 
     }
 
+    /**
+     * Gets the abridged account data from abridged specific services (abridged-api).
+     * @param resourceId - the abridged account resource link
+     * @return DocumentInfo object containing information requested from document-generator-core
+     */
+    private DocumentInfo getAbridgedAccountData(String resourceId) {
+        Accounts accounts = accountsService.getAccounts(resourceId);
+
+        if (accounts == null) {
+            return null;
+        }
+
+        AccountType accountsType = getAccountType(accounts);
+        if (accountsType == null) {
+            return null;
+        }
+
+        return new DocumentInfo();
+    }
+
+    /**
+     * Get the account type from the account link within links
+     *
+     * @return accountsData - accounts data
+     */
+    private AccountType getAccountType(Accounts accountsData) {
+        return accountsData.getLinks().keySet()
+                .stream()
+                .filter(e -> !e.equalsIgnoreCase("self"))
+                .map(AccountType::getAccountType)
+                .findFirst()
+                .orElseGet(() -> {
+                    LOG.error("Unable to find account type in account data: " + accountsData.getId());
+                    return null;
+                });
```

Comment:
if this is an error why are you returning null?

Reply:
What do you suggest returning?

Reply:
Actually this isn't an error in my opinion so I will change it to log it as debug instead :+1: